src/cockpit/packages.py

```diff
@@ -159,12 +184,18 @@ def filename_variants(filename, locale):
 
     def negotiate_file(self, path, headers):
         dirname, sep, filename = path.rpartition('/')
-        locale = headers.get('Accept-Language', '').split(',')[0].strip()  # obviously fix this
-
-        for variant in self.filename_variants(filename, locale):
-            logger.debug('consider variant %s', variant)
-            if Path(f'{dirname}{sep}{variant}') in self.files:
-                return f'{dirname}{sep}{variant}'
+        locales = []
+        accept_language = headers.get('Accept-Language')
+        if accept_language is not None:
+            locales = parse_accept_language(accept_language)
+
+        for locale in locales:
```

Comment:
This logic is definitely some kind of wrong.

 - calling the variant function multiple times means that you consider the untranslated filename over and over again
 - doing a separate scan for each page means that we can end up with a mix of languages

The C bridge tries to set the language to the first supported locale it finds, then ignores all of the others.  I wonder if we should do something like that.

If not, and you really want to check multiple languages, then I'd rather modify the filename_variants function to take a list of locales and iterate over them internally before trying the untranslated variants.